### PR TITLE
Swap 2153 upgrade material UI and use rangepicker

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

- Upgrade of `material-ui` from v4 to v5
- Using `DateRangePicker` together with `DateTimePicker` to create a custom `DateTimeRangePicker` component.
- Also created a `formik` wrapper for the `DateTimeRangePicker` component so it could be used in the forms as a formik field.
- Upgraded `@material-table/core` to version compatible with MUI 5.

## Motivation and Context

To be able to use and develop `DateTimeRangePicker` it was recommended to upgrade to MUI 5 because the `material-ui-pickers` package is deprecated(https://github.com/mui-org/material-ui-pickers/issues/2157) and now pickers are part of `@mui/lab` official package.
 
## How Has This Been Tested

- e2e tests
- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-2153

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: cypress/integration/equipments.spec.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ context('Equipment tests', () => {
 
       cy.contains(newEquipment.name)
         .parent()
-        .find('[title="View equipment"]')
+        .find('[data-testid="VisibilityIcon"]')
```
Comment (jekabs-karklins) [inline cypress/integration/equipments.spec.ts:67]: why aren't we using  data-cy here?

File: cypress/integration/scheduled-events-table.spec.ts [hunk 2]
```diff
@@ -204,26 +204,24 @@ context('Scheduled events table tests', () => {
 
       cy.contains(newScheduledEvent1.endsAt)
         .parent()
-        .find('[title="View event"]')
+        .find('[data-testid="VisibilityIcon"]')
```
Comment (jekabs-karklins) [inline cypress/integration/scheduled-events-table.spec.ts:207]: And here

**Review by jekabs-karklins — APPROVED** [review]
👍🏻

File: cypress/integration/scheduled-events-table.spec.ts (hunk 2 above)
Comment (martin-trajanovski) [reply cypress/integration/scheduled-events-table.spec.ts:207]: I think I can put our own data-cy but as long as this one exists not sure if it is needed.

**Pull request merged**